builtin/credential/chef-node/path_environment.go

```diff
@@ -0,0 +1,107 @@
+package chefNode
+
+import (
+	"strings"
+
+	"github.com/hashicorp/vault/helper/policyutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathEnvironmentsList(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "environments/?$",
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ListOperation: b.pathEnvironmentList,
+		},
+	}
+}
+
+func pathEnvironments(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: `environment/(?P<name>.+)`,
```

Comment:
I think it's fine where it is; `environments` is similar to `groups` for LDAP or `roles` for other backends.